typing/typecore.ml

```diff
@@ -5014,6 +5014,7 @@ let report_error env ppf = function
        longident lid expected provided
   | Label_mismatch(lid, trace) ->
       report_unification_error ppf env trace
+        ~explain_mismatch:false
```

Comment:
If I understand correctly, this commit disables *all* explanations for mismatched labels (ie., wrongly used record fields). This is much more than what the commit message says, and I'm not sure why this was done at all. Do you have a example where the explanation end up wrong for record fields?

Reply:
See the second part of my first message and the second quoted block: it happens in a test of the testsuite. As I said midly, "it feels a bit hackish".